Include/internal/pycore_dict.h

```diff
@@ -22,6 +22,8 @@ typedef struct {
  */
 Py_ssize_t _Py_dict_lookup(PyDictObject *mp, PyObject *key, Py_hash_t hash, PyObject **value_addr);
 
+/* Steals key and value */
+int _PyDict_SetItem(PyObject *op, PyObject *key, PyObject *value);
```

Comment:
I know that I suggested `_PyDict_SetItem` but can we make it clearer that it consumes references to `key` and `values`.

BTW I don't like the term "steal". That suggests that the caller is unaware that the references are taken, but it has to know that. I prefer "consume", or "take". Rust uses "take", IIRC.

So how about `_PyDict_SetItem_Take`, or even `_PyDict_SetItem_Takes2` to show that two references are taken.